templates/federated_reporting/mrproper.py

```diff
@@ -34,25 +37,57 @@
 if not os.geteuid() == 0:
     sys.exit("\n{} must be run as root".format(os.path.basename(__file__)))
 
-# allow environment override of cfengine WORKDIR
-WORKDIR = os.environ.get("WORKDIR", "/var/cfengine")
+WORKDIR = None
+CFE_FR_TABLES = None
+# get WORKDIR and CFE_FR_TABLES from config.sh
+config_sh_path = os.path.join(os.path.dirname(__file__), "config.sh")
+cmd = "source {}; echo $WORKDIR; echo $CFE_FR_TABLES".format(config_sh_path)
+with subprocess.Popen(
+    cmd, stdout=subprocess.PIPE, shell=True, executable="/bin/bash"
+) as proc:
+    lines = proc.stdout.readlines()
+    WORKDIR = lines[0].decode().strip()
+    CFE_FR_TABLES = lines[1].decode().strip()
+
+
+if not WORKDIR or not CFE_FR_TABLES:
+    print("Unable to get WORKDIR and CFE_FR_TABLES values from config.sh")
+    sys.exit(1)
+
+parser = argparse.ArgumentParser(
+    description="Clean up migrating clients in Federated Reporting setup"
+)
+group = parser.add_mutually_exclusive_group()
+group.add_argument("--debug", action="store_true")
+group.add_argument("--inform", action="store_true")
+args = parser.parse_args()
+
+logger = logging.getLogger("mrproper")
+ch = logging.StreamHandler()
+if args.debug:
+    logger.setLevel(logging.DEBUG)
+    ch.setLevel(logging.DEBUG)
+if args.inform:
+    logger.setLevel(logging.INFO)
+    ch.setLevel(logging.INFO)
+logger.addHandler(ch)
```

Comment:
AFAICT, this whole new block should go to the `main` function.

Reply:
Everything but the WORKDIR and env grabbing could go inside main, but I need WORKDIR in the following other global all-caps "constant" variables.